Controller/CRUDController.php

```diff
@@ -42,16 +42,22 @@ class CRUDController extends Controller
      * @param mixed   $data
      * @param integer $status
      * @param array   $headers
+     * @param Request $request
      *
      * @return Response with json encoded data
      */
-    protected function renderJson($data, $status = 200, $headers = array())
+    protected function renderJson($data, $status = 200, $headers = array(), Request $request = null)
     {
+        // to be BC
+        if (null === $request) {
```

Comment:
Cannot we have a private method ``getInternalRequest(Request $request = null)`` to avoid code duplication ?